### ISSUE TITLE
Ensures landmarks are unique

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#3835]
## Summary

Landmarks should have a unique role or role/label/title (i.e. accessible name) combination

## Locations

https://github.com/GCTC-NTGC/gc-digital-talent/blob/759d780c3dbde4012aac0b96515f1b5dacb0334d/frontend/common/src/components/NavMenu/NavMenu.tsx#L29-L55

**Comment by petertgiles** [comment]
don't forget translation
